core/evalir.ml

```diff
@@ -245,11 +245,12 @@ struct
         with NotFound _ ->
           raise (internal_error ("Failed to find function name: " ^ (string_of_int f)))
       in
-      let env =
+      let env, xs =
         match z, fvs with
-        | Some z, Some fvs -> Value.Env.bind z (fvs, Scope.Local) env
-        | None, None -> env
-        | _, _ -> assert false
+        | Some z, Some fvs -> Value.Env.bind z (fvs, Scope.Local) env, xs
+        | None, None -> env, xs
+        | Some z, None -> env, z::xs (* kludge to deal with possible mismatch *)
+        | None, Some _ -> assert false
```

Comment:
This was an attempt to work around an apparent bug in closure conversion which surfaces if you have local client functions with non-empty closures.  However I'm, sure it is not the right way to fix this problem, so I will undo it.  The right thing to do is merge the rest of this PR and then make a new issue describing the bug that needs to be worked around.